### PR TITLE
Calculate and display the accessibility level of a stop

### PR DESCRIPTION
**Pull request by Leitsi**
This change is [Reviewable](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825)

### CODE REVIEW

**Review by HenrikHartiala — CHANGES_REQUESTED** [review]
> Reviewed 5 of 5 files at r1, 9 of 9 files at r2, 2 of 2 files at r3, 3 of 3 files at r4, 7 of 7 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825)* status: all files reviewed, 7 unresolved discussions (waiting on @Leitsi)

___
*[`ui/src/components/stop-registry/stops/stop-details/layout/ExpandableInfoContainer.tsx` line 58 at r5](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz39Ygu61jIDDPTuyAd:-Nz39Ygv7yviX3ltgGEi:b18xq1f) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/87941ef32d4e70f96d64578faf968bfae3ed73ba/ui/src/components/stop-registry/stops/stop-details/layout/ExpandableInfoContainer.tsx#L58)):*
> ```XML
>         `}
>       >
>         {typeof title === 'string' ? <h4>{title}</h4> : title}
> ```

NIT, would be more safe to check if the title parameter is JSX and then do the ternary based on that. 
const isTitleJSX = React.isValidElement(title);

But we probably won't ever have any titles that are numbers etc. So doesn't matter that much :D 
___
*[`ui/src/components/stop-registry/stops/stop-details/measurements/MeasurementsSection.tsx` line 118 at r5](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz38AIZ7qaw0cuLlp0Q:-Nz38AIZ7qaw0cuLlp0R:b3eizke) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/87941ef32d4e70f96d64578faf968bfae3ed73ba/ui/src/components/stop-registry/stops/stop-details/measurements/MeasurementsSection.tsx#L118)):*
> ```XML
>         <div className="flex items-center">
>           <h4>{t('stopDetails.measurements.title')}</h4>
>           <Visible visible>
> ```

`<Visible>` which is always visible?  Probably unnecessary?
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts` line 52 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2X1jUEdoQYcQMNH6o:-Nz2X1jUEdoQYcQMNH6p:bs64cv9) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts#L52)):*
> ```XML
>     StopRegistryShelterType.Wooden,
>   ];
>   // Note: having an accessible shelter also implies that the stop has a bench, which is required for accessibility as well.
> ```

So does this mean that we do not need to check the bench attribute separately? Or is this more like "TODO: add bench checking after it is implemented", or what does this comment mean?
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts` line 53 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2YOUX7zyn94ww8Xvi:-Nz2YOUX7zyn94ww8Xvj:bpt8ll1) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts#L53)):*
> ```XML
>   ];
>   // Note: having an accessible shelter also implies that the stop has a bench, which is required for accessibility as well.
>   const shelterIsAccessible = (
> ```

This could probably be just a variable, but the function is also fine :+1: 
But would change the name to `isShelterAccessible`.
Up to you
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts` line 64 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2rGEm4W-U6TyXtSV3:-Nz2rGEm4W-U6TyXtSV4:bhwvku8) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts#L64)):*
> ```XML
> 
>   const getShelterWithAccessibilityPropertiesSet = (
>     stop: NonNullable<StopAccessibilityLevelCalculationProperties>,
> ```

It seems that we are only using a very particular part of the stop details, so could we just give the shelterEquipment as parameter, rather than the whole stop?
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts` line 75 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2qpf-0o74XXq8d3FV:-Nz2qpf-0o74XXq8d3FW:bhs2q0t) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts#L75)):*
> ```XML
> 
>   const calculateStopAccessibilityLevel = (
>     stop: StopAccessibilityLevelCalculationProperties | undefined,
> ```

Same thing here: It seems that we are only using the stop.accessibilityAssessment.hslAccessibilityProperties in this function, so should we rather just give that as parameter and not the whole stop?
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts` line 105 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2sWRi-yzxZ1dGFP1N:-Nz2sWRi-yzxZ1dGFP1O:bfesfww) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts#L105)):*
> ```XML
>     // };
>     const limits = busLimits;
>     const elevation = stopElevationFromSidewalk;
> ```

I guess this here to have the if-clauses neatly all on 1 line, which is nice, as long as it is unambiguous, which I believe it is :ok_hand: 
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.spec.ts` line 12 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2zHA-BrMyZnvy6wEv:-Nz2zHA-BrMyZnvy6wEw:bjs2g2l) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.spec.ts#L12)):*
> ```XML
> 
> describe('calculateStopAccessibilityLevel', () => {
>   // Helps readability if we can use helper functions and have most tests as oneliners.
> ```

I think that the calculateAccessibilityLevel and createStop helperfunctions are nice, but I think the rest helper functions does a bit of the opposite. 

So the `testWithMeasurements` and `testWithShelter` and this third helper function which is created first in each nested describe would be nice functions in a "normal code". But in these tests they at least for me made the readability of the test really masked and hard to follow.

It is true that this:
```
it('returns FullyAccessible when width is over 1.5 m', () => {
      testServiceWidth(150.5, StopAccessibilityLevel.FullyAccessible);
    });

```
Is pretty neat piece of code. But to see what this _actually_ does, I had to jump around and connect the pieces.
I do understand that once you understand one test with the helper functions, you understand all the rest as well. But this way it takes more time and effort to understand how this works and usually you do not need to understand all the tests, just the ones that fail.

For me this would be way more understandable:
```
      
    it('returns FullyAccessible when width is over 1.5 m', () => {
      const stop = createStop(
        {
          ...fullyAccessibleMeasurements,
          serviceAreaWidth: 150.5,
        },
        fullyAccessibleShelterEquipment,
      );

      const level = calculateAccessibilityLevel(stop);

      expect(level).toEqual(StopAccessibilityLevel.FullyAccessible);
    });
```
Just by looking at this, I can see that we create a stop that has otherwise fully accessible measurements, but we override one measurement, then we calculate the level and then I can clearly see what is expected. And at least for me it is more important to have one particular test easy to read and understand, than to have it easy to write more tests.
This also follows the principles of The AAA pattern (Arrange, Act, Assert).

But **all* this said, Im not going to put this as a blocking comment since I know this is how many want to create tests, but I believe it is not the best way and if the reasons I gave would be enough to turn the ship around, I would be really happy :D
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.spec.ts` line 274 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz35pO2DBvSUqhATAv4:-Nz35pO2DBvSUqhATAv5:b-pi6scm) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.spec.ts#L274)):*
> ```XML
>     it('returns FullyAccessible if all required accessibility properties are acceptable and other measurements are within limits', () => {
>       const stop = createStop(
>         {
> ```

Unnecessary spread, probably left from copypaste



_Suggestion:_
```XML
      const stop = createStop(
        fullyAccessibleMeasurements,
        fullyAccessibleShelterEquipment,
      );
```

**Review by Leitsi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825)* status: all files reviewed, 6 unresolved discussions (waiting on @HenrikHartiala)

___
*[`ui/src/components/stop-registry/stops/stop-details/layout/ExpandableInfoContainer.tsx` line 58 at r5](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz39Ygu61jIDDPTuyAd:-Nz6zQcXBlQW_lbJmMUe:b-3wk2u3) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/87941ef32d4e70f96d64578faf968bfae3ed73ba/ui/src/components/stop-registry/stops/stop-details/layout/ExpandableInfoContainer.tsx#L58)):*
<details><summary><i>Previously, HenrikHartiala wrote…</i></summary><blockquote>

NIT, would be more safe to check if the title parameter is JSX and then do the ternary based on that. 
const isTitleJSX = React.isValidElement(title);

But we probably won't ever have any titles that are numbers etc. So doesn't matter that much :D
</blockquote></details>

:thumbsup:  sounds like a more correct way to do this anyways, changed.
___
*[`ui/src/components/stop-registry/stops/stop-details/measurements/MeasurementsSection.tsx` line 118 at r5](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz38AIZ7qaw0cuLlp0Q:-Nz6ztslEtQRN6iz5zUF:b-u4jgr8) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/87941ef32d4e70f96d64578faf968bfae3ed73ba/ui/src/components/stop-registry/stops/stop-details/measurements/MeasurementsSection.tsx#L118)):*
<details><summary><i>Previously, HenrikHartiala wrote…</i></summary><blockquote>

`<Visible>` which is always visible?  Probably unnecessary?
</blockquote></details>

Yep, forgot this from some earlier version, removed.
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts` line 52 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2X1jUEdoQYcQMNH6o:-Nz7-AqT44zW2qMN852U:batxlg7) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts#L52)):*
<details><summary><i>Previously, HenrikHartiala wrote…</i></summary><blockquote>

So does this mean that we do not need to check the bench attribute separately? Or is this more like "TODO: add bench checking after it is implemented", or what does this comment mean?
</blockquote></details>

Nope.
The definitions we got from HSL stated that high accessibility requires that a stop has shelter and a bench. We don't have bench in our data model, nor are there plans to add it. But, shelter _implies_ that there is a bench, since shelters always have a bench (most types at least, listed in `accessibleShelterTypes` here).

I did some rewording here.
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts` line 53 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2YOUX7zyn94ww8Xvi:-Nz7-gsOCKfGCN0Ft15x:bauwoid) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts#L53)):*
<details><summary><i>Previously, HenrikHartiala wrote…</i></summary><blockquote>

This could probably be just a variable, but the function is also fine :+1: 
But would change the name to `isShelterAccessible`.
Up to you
</blockquote></details>

:thumbsup:  renamed.
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts` line 64 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2rGEm4W-U6TyXtSV3:-Nz70Uwh77XoiSCB1k8Y:b-k6zi00) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts#L64)):*
<details><summary><i>Previously, HenrikHartiala wrote…</i></summary><blockquote>

It seems that we are only using a very particular part of the stop details, so could we just give the shelterEquipment as parameter, rather than the whole stop?
</blockquote></details>

Could, but with this version the caller doesn't have to care that much what properties are actually used for calculating the stop accessibility level.
So now we can just do:
```
interface Props {
  stop: StopWithDetails;
}

const accessibilityLevel = calculateStopAccessibilityLevel(stop.stop_place);
```
but with more specific params this would have to be something like:
```
const calculateStopAccessibilityLevel = (
  hslAccessibilityProperties?: HslAccessibilityLevelCalculationProperties,
  shelterEquipment?: Array<ShelterAccessibilityLevelCalculationProperties> | null
) => {
}

const accessibilityLevel = calculateStopAccessibilityLevel(
  stop.stop_place?.accessibilityAssessment?.hslAccessibilityProperties,
  stop.stop_place?.quays?.[0]?.placeEquipments?.shelterEquipment || []
);
```

This kind of API requires a bit more setup in the unit tests, but otherwise it's worth it I think.

Could have maybe added some kind of helper function to the hook to handle this property picking, but there's no payback in doing that at this point.
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts` line 75 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2qpf-0o74XXq8d3FV:-Nz72H9h1nHkOukexUm7:bsjte78) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts#L75)):*
<details><summary><i>Previously, HenrikHartiala wrote…</i></summary><blockquote>

Same thing here: It seems that we are only using the stop.accessibilityAssessment.hslAccessibilityProperties in this function, so should we rather just give that as parameter and not the whole stop?
</blockquote></details>

See above comment.
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts` line 105 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2sWRi-yzxZ1dGFP1N:-Nz72QTM244PpXp_IKEe:b2xj1e4) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts#L105)):*
<details><summary><i>Previously, HenrikHartiala wrote…</i></summary><blockquote>

I guess this here to have the if-clauses neatly all on 1 line, which is nice, as long as it is unambiguous, which I believe it is :ok_hand:
</blockquote></details>

Also some preparation for the future. With trams this needs to be changed to something like:
```
const limits = isTram ? tramLimits : busLimits;
const elevation = isTram ? stopElevationFromRailTop : stopElevationFromSidewalk;
```
And the actual if clauses below will remain pretty much unchanged.
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.spec.ts` line 12 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2zHA-BrMyZnvy6wEv:-Nz73Ej-DSazWxPNoLHy:b9elk1y) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.spec.ts#L12)):*
> But to see what this _actually_ does, I had to jump around and connect the pieces.

This is indeed _exactly_ what should usually be avoided, so yep I've failed a bit here.

I think what goes wrong here is that you have to jump through _two_ helper functions.
Eg. this I think would still be quite fine:
```
  describe('serviceAreaWidth', () => {
    const testServiceWidth = (
      width: (typeof fullyAccessibleMeasurements)['serviceAreaWidth'],
      expectedLevel: StopAccessibilityLevel,
    ) => {
      const stop = createStop(
        {
          ...fullyAccessibleMeasurements,
          serviceAreaWidth: width,
        },
        fullyAccessibleShelterEquipment,
      );

      const level = calculateAccessibilityLevel(stop);

      expect(level).toEqual(expectedLevel);
    };

    it('returns FullyAccessible when width is over 1.5 m', () => {
      testServiceWidth(150.5, StopAccessibilityLevel.FullyAccessible);
    });

    it('returns FullyAccessible when width is exactly 1.5 m', () => {
      testServiceWidth(150, StopAccessibilityLevel.FullyAccessible);
    });
    // ...
```
since here you can still easily see the actual test code, without jumping around too much.

Think I'll remove some of these "helper" functions and copypaste closer to actual tests, in the manner I just described.
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.spec.ts` line 274 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz35pO2DBvSUqhATAv4:-Nz74gn1CkjpMq-3Pfib:b-896fix) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.spec.ts#L274)):*
<details><summary><i>Previously, HenrikHartiala wrote…</i></summary><blockquote>

Unnecessary spread, probably left from copypaste
</blockquote></details>

Done.

**Review by HenrikHartiala — APPROVED** [review]
> Reviewed 3 of 17 files at r6, 30 of 30 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @Leitsi)

___
*[`ui/src/components/stop-registry/stops/stop-details/layout/ExpandableInfoContainer.tsx` line 58 at r5](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz39Ygu61jIDDPTuyAd:-NzBwFZU5GUaShsdSu3B:b4putsz) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/87941ef32d4e70f96d64578faf968bfae3ed73ba/ui/src/components/stop-registry/stops/stop-details/layout/ExpandableInfoContainer.tsx#L58)):*
<details><summary><i>Previously, Leitsi (Tommi Leinamo) wrote…</i></summary><blockquote>

:thumbsup:  sounds like a more correct way to do this anyways, changed.
</blockquote></details>

:ok_hand: 
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts` line 52 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2X1jUEdoQYcQMNH6o:-NzBwkXc2Z2VV1Ym2R6z:b-hf540s) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts#L52)):*
<details><summary><i>Previously, Leitsi (Tommi Leinamo) wrote…</i></summary><blockquote>

Nope.
The definitions we got from HSL stated that high accessibility requires that a stop has shelter and a bench. We don't have bench in our data model, nor are there plans to add it. But, shelter _implies_ that there is a bench, since shelters always have a bench (most types at least, listed in `accessibleShelterTypes` here).

I did some rewording here.
</blockquote></details>

Ah thanks, now I understand. And good rewording also :ok_hand: 
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts` line 64 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2rGEm4W-U6TyXtSV3:-NzByzbm-FDgoVZAJWxp:b-6e7sp4) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts#L64)):*
<details><summary><i>Previously, Leitsi (Tommi Leinamo) wrote…</i></summary><blockquote>

Could, but with this version the caller doesn't have to care that much what properties are actually used for calculating the stop accessibility level.
So now we can just do:
```
interface Props {
  stop: StopWithDetails;
}

const accessibilityLevel = calculateStopAccessibilityLevel(stop.stop_place);
```
but with more specific params this would have to be something like:
```
const calculateStopAccessibilityLevel = (
  hslAccessibilityProperties?: HslAccessibilityLevelCalculationProperties,
  shelterEquipment?: Array<ShelterAccessibilityLevelCalculationProperties> | null
) => {
}

const accessibilityLevel = calculateStopAccessibilityLevel(
  stop.stop_place?.accessibilityAssessment?.hslAccessibilityProperties,
  stop.stop_place?.quays?.[0]?.placeEquipments?.shelterEquipment || []
);
```

This kind of API requires a bit more setup in the unit tests, but otherwise it's worth it I think.

Could have maybe added some kind of helper function to the hook to handle this property picking, but there's no payback in doing that at this point.
</blockquote></details>

Ah yeah, makes sense. I realised that this does not **require** the whole stop as parameter, which is nice. (We've had some of those earlier).
Although this helper function (getShelterWith..) seems to still require the `accessibilityAssessment` as parameter, even though it does not use it. But thats not a big deal so all OK.
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts` line 105 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2sWRi-yzxZ1dGFP1N:-NzC4A-p0CcOaw6XHG2N:be0vjdf) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts#L105)):*
<details><summary><i>Previously, Leitsi (Tommi Leinamo) wrote…</i></summary><blockquote>

Also some preparation for the future. With trams this needs to be changed to something like:
```
const limits = isTram ? tramLimits : busLimits;
const elevation = isTram ? stopElevationFromRailTop : stopElevationFromSidewalk;
```
And the actual if clauses below will remain pretty much unchanged.
</blockquote></details>

Ah, ok I see :+1: nice.
___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.spec.ts` line 12 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-Nz2zHA-BrMyZnvy6wEv:-NzC4HAlCIXGlUV-4pw-:b-pqc5io) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/6d24468b5619a6a76d133b82a4de19a88f133de3/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.spec.ts#L12)):*
<details><summary><i>Previously, Leitsi (Tommi Leinamo) wrote…</i></summary><blockquote>

> But to see what this _actually_ does, I had to jump around and connect the pieces.

This is indeed _exactly_ what should usually be avoided, so yep I've failed a bit here.

I think what goes wrong here is that you have to jump through _two_ helper functions.
Eg. this I think would still be quite fine:
```
  describe('serviceAreaWidth', () => {
    const testServiceWidth = (
      width: (typeof fullyAccessibleMeasurements)['serviceAreaWidth'],
      expectedLevel: StopAccessibilityLevel,
    ) => {
      const stop = createStop(
        {
          ...fullyAccessibleMeasurements,
          serviceAreaWidth: width,
        },
        fullyAccessibleShelterEquipment,
      );

      const level = calculateAccessibilityLevel(stop);

      expect(level).toEqual(expectedLevel);
    };

    it('returns FullyAccessible when width is over 1.5 m', () => {
      testServiceWidth(150.5, StopAccessibilityLevel.FullyAccessible);
    });

    it('returns FullyAccessible when width is exactly 1.5 m', () => {
      testServiceWidth(150, StopAccessibilityLevel.FullyAccessible);
    });
    // ...
```
since here you can still easily see the actual test code, without jumping around too much.

Think I'll remove some of these "helper" functions and copypaste closer to actual tests, in the manner I just described.
</blockquote></details>

As discussed in slack, this is a nice solution. Thanks!

**Review by HenrikHartiala — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @Leitsi)

___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts` line 66 at r7](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-NzC50xP4BXuVhd0j_QG:-NzC50xP4BXuVhd0j_QH:bn0yjq3) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/0678fd3ae40ca8503a00ca1d27a1fdb3a378390c/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts#L66)):*
> ```XML
>   };
> 
>   const getShelterWithAccessibilityPropertiesSet = (
> ```

Ahh, btw. `extract` could be better prefix here than `get` since `get` usually implies that something is fetched. But just a suggestion and I think we have used both anyways.

**Review by Leitsi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @Leitsi)

___
*[`ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts` line 66 at r7](https://reviewable.io/reviews/HSLdevcom/jore4-ui/825#-NzC50xP4BXuVhd0j_QG:-NzCEY652Pg7FPF_HJ7f:b-nmpjk7) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/0678fd3ae40ca8503a00ca1d27a1fdb3a378390c/ui/src/hooks/stop-registry/useCalculateStopAccessibilityLevel.ts#L66)):*
<details><summary><i>Previously, HenrikHartiala wrote…</i></summary><blockquote>

Ahh, btw. `extract` could be better prefix here than `get` since `get` usually implies that something is fetched. But just a suggestion and I think we have used both anyways.
</blockquote></details>

True, `get` is a bit ambiguous and can mean pretty much anything. Changed to `find` since that's what we're actually doing here.
